### ISSUE TITLE
Add an authentication server for self-hosted users who do not use Firebase

### ISSUE THREAD

**Issue by alex4108** [successible/cleanslate#61]
Hihi, hope you're well today.

I'm working to deploy this on my k8s homelab (I do plan to publish the repo when done!) and am concerned about a potentially insecure default.

The way I read it, [deploy.sh](https://github.com/successible/cleanslate/blob/main/deploy.sh) defines the `HASURA_GRAPHQL_JWT_SECRET` as a static UUID for the value?  Is this not an exposed secret that potentially the world could use to compromise an environment?

It looks even weirder when I look in [getJWT.ts](https://github.com/successible/cleanslate/blob/main/src/helpers/getJWT.ts) and see the static SIGNING_KEY used in non-firebase implementations. The `HASURA_GRAPHQL_JWT_SECRET` value isn't even used here?

Please let me know if I should change the "key" value for my deployments to secure them.

Many thanks.

**Comment by paulzakin** [comment]
So, Hasura only accepts a JWT. For Firebase deployments, that is not a problem. We just use the Firebase's JWT set by a private HASURA_GRAPHQL_JWT_SECRET in a private `.env` file. Very secure!

For self-hosted deploys, we want people to just use an API token. Hence, we "sign" your API token to make it a JWT so that Hasura can consume it. Hasura then returns all data that matches your secret API token. 

Hence, in theory, someone could sign your API token to get data from your instance. However, they would have to know your API token, which is a uuid, a long, randomly generated value. Effectively, not guessable.

So, the JWT is insecure on self-hosted deployments (correct!) but the API token it holds is not. And that is what returns data.

Basically, it is not different that just passing a long, API token on any other site to get your data. We just have to wrap it in this insecure JWT to make Hasura happy. 

Hopefully, that makes sense!

**Comment by paulzakin** [comment]
However, thinking more about it - now that we have enough interest in self-hosting without Firebase, I'm willing to create a little server that handles the JWT creation to avoid this problem. You are likely not the last person with this question, and I only created this hacky API token when very recently people started asking me for a solution without Firebase. And while I feel it is secure, it could one, be more secure, and two, not violate the principle of least astonishement.

**Comment by paulzakin** [comment]
Self-hosted versions of Clean Slate that do not use Firebase now have an authentication server spun up with Docker Compose. This eliminates the hacky solution outline above, and every self-hosted instance can now get their own signing key!
